docs/docs/reference/debugger/debugger_repl.md

```diff
@@ -0,0 +1,388 @@
+
+#### `continue` (c)
+
+Continues execution until the next breakpoint, or the end of the program.
+
+#### `restart` (res)
+
+Interrupts execution, and restarts a new debugging session from scratch.
+
+#### `opcodes` (o)
+
+Display the program's ACIR opcode sequence. For example:
+
+```
+0  BLACKBOX::RANGE [(_0, num_bits: 32)] [ ]
+1 ->  BRILLIG inputs=[Single(Expression { mul_terms: [], linear_combinations: [(1, Witness(0))], q_c: 0 })] outputs=[Simple(Witness(1))]
+	1.0  |   Mov { destination: RegisterIndex(2), source: RegisterIndex(0) }
+	1.1  |   Const { destination: RegisterIndex(0), value: Value { inner: 0 } }
+	1.2  |   Const { destination: RegisterIndex(1), value: Value { inner: 0 } }
+	1.3  |   Mov { destination: RegisterIndex(2), source: RegisterIndex(2) }
+	1.4  |   Call { location: 7 }
+	...
+	1.43 |   Return
+2     EXPR [ (1, _1) -2 ]
+```
+
+### Breakpoints
+
+#### `break [Opcode]` (or shorthand `b [Opcode]`)
+
+Sets a breakpoint on the specified opcode index. To get a list of the program opcode numbers, see [the `opcode` command](#opcodes-o). For example:
+
+```
+0  BLACKBOX::RANGE [(_0, num_bits: 32)] [ ]
+1 ->  BRILLIG inputs=[Single(Expression { mul_terms: [], linear_combinations: [(1, Witness(0))], q_c: 0 })] outputs=[Simple(Witness(1))]
+	1.0  |   Mov { destination: RegisterIndex(2), source: RegisterIndex(0) }
+	1.1  |   Const { destination: RegisterIndex(0), value: Value { inner: 0 } }
+	1.2  |   Const { destination: RegisterIndex(1), value: Value { inner: 0 } }
+	1.3  |   Mov { destination: RegisterIndex(2), source: RegisterIndex(2) }
+	1.4  |   Call { location: 7 }
+	...
+	1.43 |   Return
+2    EXPR [ (1, _1) -2 ]
+```
+
+In this example, issuing a `break 1.2` commmand adds break on opcode 1.2, as denoted by the `*` character:
+
+```
+0  BLACKBOX::RANGE [(_0, num_bits: 32)] [ ]
+1 ->  BRILLIG inputs=[Single(Expression { mul_terms: [], linear_combinations: [(1, Witness(0))], q_c: 0 })] outputs=[Simple(Witness(1))]
+	1.0  |   Mov { destination: RegisterIndex(2), source: RegisterIndex(0) }
+	1.1  |   Const { destination: RegisterIndex(0), value: Value { inner: 0 } }
+	1.2  | * Const { destination: RegisterIndex(1), value: Value { inner: 0 } }
+	1.3  |   Mov { destination: RegisterIndex(2), source: RegisterIndex(2) }
+	1.4  |   Call { location: 7 }
+	...
+	1.43 |   Return
+2    EXPR [ (1, _1) -2 ]
+```
+
+Running [the `continue` command](#continue-c) at this point would cause the debugger to execute the program until opcode 1.2.
+
+#### `delete [Opcode]` (or shorthand `d [Opcode]`)
+
+Deletes a breakpoint at an opcode location. Usage is analogous to [the `break` command](#).
+
+### Variable inspection
+
+#### vars
+
+Show variable values available at this point in execution.
+
+Note: that the ability to inspect variable values from the debugger depends on compilation to be run in a special debug instrumentation mode. This instrumentation weaves variable tracing code with the original source code. 
```

Comment:
```suggestion
:::note

The ability to inspect variable values from the debugger depends on compilation to be run in a special debug instrumentation mode. This instrumentation weaves variable tracing code with the original source code. 

:::
```

Using admonitions block is highly encouraged

Reply:
Done!